### PR TITLE
CI: Reuse check schema

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of #833.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [n/a] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
Blocked by https://github.com/Lightning-AI/devtools/pull/15/.

File: .github/workflows/ci-schema.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+on:
+  push:
+    branches: [master, "release/*"]
+  pull_request:
+    branches: [master, "release/*"]
+
+jobs:
+  check:
+    uses: Lightning-AI/devtools/.github/workflows/check-schema.yml@main  # FIXME: Use v0.1.0
```
Comment (akihironitta) [inline .github/workflows/ci-schema.yml:11]: TODO: Apply this suggestion once https://github.com/Lightning-AI/devtools/pull/15 is merged.
```suggestion
    uses: Lightning-AI/devtools/.github/workflows/check-schema.yml@v0.1.0
```

**Comment by akihironitta** [comment]
Hi @Borda would you mind having a quick look?

File: .github/workflows/ci-schema.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+on:
+  push:
+    branches: [master, "release/*"]
+  pull_request:
+    branches: [master, "release/*"]
+
+jobs:
+  validate-schema:
+    uses: Lightning-AI/devtools/.github/workflows/check-schema.yml@v0.1.0
```
Comment (Borda) [inline .github/workflows/ci-schema.yml:11]:
```suggestion
    uses: Lightning-AI/utilities/.github/workflows/check-schema.yml@v0.1.0
```

**Review by Borda — APPROVED** [review]

**Comment by akihironitta** [comment]
This PR also requires action from admin since one of the required jobs changed in naming.

**Review by otaj — APPROVED** [review]

**Pull request merged**